### ISSUE TITLE
Validate user-provided values before use

### ISSUE THREAD

**Issue by basbossink** [magit/magit#2901]
Recently I upgraded magit to version: 20161214.521 since then I can no longer perform a commit using magit, the git process buffer reads:
```
1 C:/Users/user/AppData/Local/Programs/Git/mingw64/libexec/git-core/git … commit --

*ERROR*: Format specifier doesn’t match argument typeerror: There was a problem with the editor '"c:/Users/user/PortableApps/emacs/bin/emacsclient.exe"'.
Please supply the message using either -m or -F option.
```

Using the procedure described in the issue template :+1:, I discovered that the problem is caused by the following line in my configuration:
```
(setq-default git-commit-summary-max-length git-commit-fill-column)
```
`C-h v git-commit-fill-column` shows it has the value `nil`, (it's default was changed in commit: 2087cce6c021c3f985e5c53caa74bf56ebdd30fe , which broke my configuration) adding the line:
```
(setq git-commit-fill-column 78)
```
fixes the problem.

My 'complaint' in this case would be that the 'error message' is not very helpful in this case. Secondly maybe it is an idea to introduce 'setter-functions' that can perform some basic validation on variables that are customizable, or make the code in 
```
(defun git-commit-summary-regexp ()
```
handle the case where the `git-commit-summary-max-length` is not set correctly.
Another suggestion could be to wrap any `-hook`s in a `condition-case` [form](https://www.gnu.org/software/emacs/manual/html_node/elisp/Handling-Errors.html), but I'm no stretch of the imagination on Emacs Lisp expert.

Since I have now 'fixed' my initial problem I'm fine with this issue being put on low priority. I also submitted this to help any people that run into similar issues.

**Comment by npostavs** [comment]
Oh, I was going to mark that variable obsolete but I forgot about it, that might have helped you catch this sooner.

> maybe it is an idea to introduce 'setter-functions' that can perform some basic validation on variables that are customizable,

Hmm, maybe: https://elpa.gnu.org/packages/validate.html

> Another suggestion could be to wrap any `-hook`s in a `condition-case` [form](https://www.gnu.org/software/emacs/manual/html_node/elisp/Handling-Errors.html)

The trouble with this idea is that it's unlikely we could come up with a better error message (given the information a condition handler receives) than what's printed by default.

**Comment by tarsius** [comment]
`validate` sounds like a good option.

**Comment by tarsius** [comment]
We haven't really had many issues that were due to users setting variables to invalid values, and I have therefore decided to not do this. Doing it would be much more work than occasionally help a user who has messed their configuration up. Even if we tried validating everything, there still would be room for setting some invalid value that we don't catch. Also there already are "setter functions", which do validation -- provided the user uses the custom interface.
